### ISSUE TITLE
vs2019 调试五轮，ep05_0.0278_0.0990_0.1268 就很难下降了

### ISSUE THREAD

**Issue by hsome2020** [opconty/pytorch_ctpn#31]
对代码做了一些改动，发现原始代码 loss 很难下降，是计算 loss 时维度出了问题，二者不一样，导致第二轮就很难收敛；我也稍做了一些改动，但到 0.0278_0.0990_0.1268 就下不去了。
想问下，你真调到了 0.07 以下吗？
ctpn_ep01_0.0411_0.1523_0.1934.pth
ctpn_ep02_0.0309_0.1167_0.1477.pth
ctpn_ep03_0.0282_0.1071_0.1353.pth
ctpn_ep04_0.0279_0.1046_0.1325.pth
ctpn_ep05_0.0278_0.0990_0.1268.pth
这是五轮结果(^_^)

**Comment by hsome2020** [comment]
那个权重需要翻墙下吗？我翻不了墙，能发我邮箱吗 308161302@qq.com, 谢谢 (^_^)

**Comment by opconty** [comment]
@hsome2020 可能需要一些调参经验，我发给你，请查收

**Comment by hsome2020** [comment]
刚接触 PYTHON, 深度学习，半个月前连变量都不会声明，你这个是我第一个真正调通的，
主要是 batch_norm 改成了 groupnorm, 另外 gru 把batch_first 变成了 false, 应该是 width 对应 seq_length, 所以 N*H 应该对 batch_size, 还有就是 把最后输出 到 loss 分了两层， 先从 512->256， 再从 256 -> 20 , 看上去快了，但后面就很难下去了，也没多 运行，最多5 个 epoch(^_^)
谢谢！！！

**Comment by opconty** [comment]
加油

**Comment by zty8599** [comment]
> @ hsome2020可能需要一些调参经验，我发给你，请查收

请问调参经验可以也发我一下吗 我的邮箱是784954586@qq.com 谢谢

**Comment by opconty** [comment]
@zty8599 经验看来只能靠自己积累了，要是能发就好了

**Comment by xieguoxuan** [comment]
> 刚接触 PYTHON, 深度学习，半个月前连变量都不会声明，你这个是我第一个真正调通的，
> 主要是 batch_norm 改成了 groupnorm, 另外 gru 把batch_first 变成了 false, 应该是 width 对应 seq_length, 所以 N*H 应该对 batch_size, 还有就是 把最后输出 到 loss 分了两层， 先从 512->256， 再从 256 -> 20 , 看上去快了，但后面就很难下去了，也没多 运行，最多5 个 epoch(^_^)
> 谢谢！！！



> 刚接触 PYTHON, 深度学习，半个月前连变量都不会声明，你这个是我第一个真正调通的，
> 主要是 batch_norm 改成了 groupnorm, 另外 gru 把batch_first 变成了 false, 应该是 width 对应 seq_length, 所以 N*H 应该对 batch_size, 还有就是 把最后输出 到 loss 分了两层， 先从 512->256， 再从 256 -> 20 , 看上去快了，但后面就很难下去了，也没多 运行，最多5 个 epoch(^_^)
> 谢谢！！！



> 刚接触 PYTHON, 深度学习，半个月前连变量都不会声明，你这个是我第一个真正调通的，
> 主要是 batch_norm 改成了 groupnorm, 另外 gru 把batch_first 变成了 false, 应该是 width 对应 seq_length, 所以 N*H 应该对 batch_size, 还有就是 把最后输出 到 loss 分了两层， 先从 512->256， 再从 256 -> 20 , 看上去快了，但后面就很难下去了，也没多 运行，最多5 个 epoch(^_^)
> 谢谢！！！

请问你最后检测结果如何呢，我用icdar2013数据集，调到0.03还是检测效果很差

**Comment by xieguoxuan** [comment]
> 刚接触 PYTHON, 深度学习，半个月前连变量都不会声明，你这个是我第一个真正调通的，
> 主要是 batch_norm 改成了 groupnorm, 另外 gru 把batch_first 变成了 false, 应该是 width 对应 seq_length, 所以 N*H 应该对 batch_size, 还有就是 把最后输出 到 loss 分了两层， 先从 512->256， 再从 256 -> 20 , 看上去快了，但后面就很难下去了，也没多 运行，最多5 个 epoch(^_^)
> 谢谢！！！

请问你最后检测结果如何呢，我用icdar2013数据集，调到0.03还是检测效果很差

**Comment by opconty** [comment]
如果很难收敛，建议参考下我的训练log，希望有所帮助

[训练日志](https://github.com/opconty/pytorch_ctpn/blob/master/logs/training_logs.pdf)

**Comment by bjlgcxc** [comment]
我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪,  供大家参考

**Comment by opconty** [comment]
@bjlgcxc 你看他的references咯

**Comment by bjlgcxc** [comment]
> @bjlgcxc 你看他的references咯

看到了那个reference，但是两个代码训练出来的效果确实差比较多，ctpn.pytorch这个代码不需要pretrain model，效果也比较好. 只是给大家参考一下，如果大家有空可以对比下看看吧

**Comment by yangh-zzf-itcast** [comment]
> 我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪, 供大家参考

您好，请问你这个工程icdar数据集训练了多久收敛的呢？另外最佳性能大概是多少

**Comment by rui-alt** [comment]
> 我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪, 供大家参考

为什么我用这个代码训icdar2015训了一百个epoch,loss下降到0.07，最来用来预测一点效果也没有

**Comment by opconty** [comment]
> > 我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪, 供大家参考
> 
> 为什么我用这个代码训icdar2015训了一百个epoch,loss下降到0.07，最来用来预测一点效果也没有



> > 我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪, 供大家参考
> 
> 为什么我用这个代码训icdar2015训了一百个epoch,loss下降到0.07，最来用来预测一点效果也没有

这个模型已经很久很久之前的了，不建议继续在这上面花费时间精力，有更好的选择哈

**Comment by rui-alt** [comment]
> > > 我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪, 供大家参考
> > 
> > 
> > 为什么我用这个代码训icdar2015训了一百个epoch,loss下降到0.07，最来用来预测一点效果也没有
> 
> > > 我尝试了https://github.com/CrazySummerday/ctpn.pytorch 这个代码, 和这个repo的代码基本没啥差异, 但是ctpn.pytorch的这个代码很快就能在数据上收敛, 暂时还没对比两个代码的详细区别在哪, 供大家参考
> > 
> > 
> > 为什么我用这个代码训icdar2015训了一百个epoch,loss下降到0.07，最来用来预测一点效果也没有
> 
> 这个模型已经很久很久之前的了，不建议继续在这上面花费时间精力，有更好的选择哈

我毕设选的这个模型，已经来不及换了呜呜
